### PR TITLE
add sbt imetering implementation

### PR DESCRIPTION
**Pull request by sagar-amberflo**
### Context

This repository is a Amberflo implementation of AWS SBT IMetering implementation.

[AWS SaaS Builder Toolkit (SBT)](https://github.com/awslabs/sbt-aws) is an open-source developer toolkit to implement SaaS best practices. It provides reusable components for control plane and application plane concepts such as tenant and user management, billing, and metering. 

Amberflo has partnered with AWS SBT to provide Metering component in AWS SBT. This repository provides a detailed implementation of the Amberflo metering service for use with the AWS SaaS Factory Software Builder Toolkit (SBT).

### Main changes

amberflo-metering.ts - Amberflo implementation of [IMetering](https://github.com/awslabs/sbt-aws/blob/main/src/control-plane/metering/metering-interface.ts#L10) component.
metering-service.py - lambda handler for the amberflo metering
README.md - documentation of the repository.

### Test
This implemention has been tested locally by deploying a sample SBT application and verifying all the available metering APIs .

### TODO

- [ ] publish to npm
- [ ] test by installing this repository from npm in sbt application.

### CODE REVIEW

File: resources/functions/metering-service.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+@app.put("/meters/<meterId>")
+@tracer.capture_method
+def update_meter():
+    payload = app.current_event.json_body
+    if not all(k in payload for k in ['label', 'meterApiName', 'meterType']):
+        raise BadRequestError("Required properties missing for update meter")
+    payload.setdefault('id', meterId)
+    return make_api_call('PUT', '/meters', payload)
+
+@app.get("/usage/meterId")
```
Comment (talha-aftab) [inline resources/functions/metering-service.py]: Should this be `"/usage/<meterId>"`?

**Review by talha-aftab — APPROVED** [review]

File: resources/functions/metering-service.py (hunk 1 above)
Comment (sagar-amberflo) [reply resources/functions/metering-service.py]: They have defined it like this https://github.com/awslabs/sbt-aws/blob/main/src/control-plane/metering/metering-provider.ts#L76 for now, they will update it to `/usage/<meterId>`.

To keep it generic their rationale was to have a GET endpoint for usage and with a path parameter for specific meter.

Comment (talha-aftab) [reply resources/functions/metering-service.py]: ok

**Review by talha-aftab — APPROVED** [review]

File: resources/functions/metering-service.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+@app.delete("/meters/<meter_id>")
+@tracer.capture_method
+def delete_meter(meter_id: Annotated[str, Path(min_length=1)]):
+    meter = make_api_call('DELETE', f'/meters/{meter_id}', None)
+    return {'data': meter}, HTTPStatus.OK
+
+@app.get("/usage/<meter_id>")
+@tracer.capture_method
+def fetch_usage(meter_id: Annotated[str, Path(min_length=1)]):
+    payload = app.current_event.query_string_parameters or {}
```
Comment (vgeorgewillv) [inline resources/functions/metering-service.py:87]: if payload = {}, what happens?

**Review by vgeorgewillv — COMMENTED** [review]
small questions

**Pull request merged**